### PR TITLE
homework.lambda.part1.exercise1 (Prodan Ivan)

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/test/java/lambda/part1/exercise/Exercise1.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+                new Person("Алексей", "Доренко", 40),
+                new Person("Артем", "Зимов", 45)
+        )));
+    }
+
+    @Test
+    void sortPersonsByAgeUsingArraysSortAnonymousComparator() {
+        Person[] persons = getPersons();
+
+        Arrays.sort(persons, Comparator.comparingInt(Person::getAge));
```
Comment (Lopat) [inline src/test/java/lambda/part1/exercise/Exercise1.java]: anonimous comparator  - means you should declare anonimous class by 
new Comparator() {
     public int compare(.....) {
           return ...;
        }
    });

File: src/test/java/lambda/part1/exercise/Exercise1.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+import java.util.Objects;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.arrayContaining;
+import static org.hamcrest.Matchers.is;
+
+class Exercise1 {
+
+    @Test
+    void sortPersonsByAgeUsingArraysSortLocalComparator() {
```
Comment (Lopat) [inline src/test/java/lambda/part1/exercise/Exercise1.java:21]: local comparator - declare local class MyComparator implements Comparator<Person>
and implement compare method

File: src/test/java/lambda/part1/exercise/Exercise1.java (hunk 1 above)
Comment (ghost) [reply src/test/java/lambda/part1/exercise/Exercise1.java]: Done.

**Review by Lopat — APPROVED** [review]

File: src/test/java/lambda/part1/exercise/Exercise1.java [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+                new Person("Алексей", "Доренко", 40),
+                new Person("Артем", "Зимов", 45)
+        )));
+    }
+
+    @Test
+    void sortPersonsByLastNameThenFirstNameUsingArraysSortAnonymousComparator() {
+        Person[] persons = getPersons();
+
+        Arrays.sort(persons, Comparator.comparing(Person::getLastName).thenComparing(Person::getFirstName));
```
Comment (vkovrovEPAM) [inline src/test/java/lambda/part1/exercise/Exercise1.java]: It is not an anonymous comparator

Comment (vkovrovEPAM) [reply src/test/java/lambda/part1/exercise/Exercise1.java]: I should do it as in previous test method

File: src/test/java/lambda/part1/exercise/Exercise1.java [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+        assertThat(person, is(new Person("Николай", "Зимов", 30)));
+    }
+
+    @SuppressWarnings("Guava")
+    @Test
+    void findFirstWithAge30UsingGuavaAnonymousPredicate() {
+        List<Person> persons = Arrays.asList(getPersons());
+
+        Optional<Person> personOptional = FluentIterable.from(persons)
+                .firstMatch(person -> Objects.equals(person.getAge(), 30));
```
Comment (vkovrovEPAM) [inline src/test/java/lambda/part1/exercise/Exercise1.java]: It is not an anonymous predicate

**Review by vkovrovEPAM — CHANGES_REQUESTED** [review]

**Review by vkovrovEPAM — APPROVED** [review]

**Pull request merged**